token/js/src/instructions/uiAmountToAmount.ts

```diff
@@ -0,0 +1,132 @@
+import { u8, struct, blob } from '@solana/buffer-layout';
+import { AccountMeta, PublicKey, TransactionInstruction } from '@solana/web3.js';
+import { TOKEN_PROGRAM_ID } from '../constants';
+import {
+    TokenInvalidInstructionDataError,
+    TokenInvalidInstructionKeysError,
+    TokenInvalidInstructionProgramError,
+    TokenInvalidInstructionTypeError,
+} from '../errors';
+import { TokenInstruction } from './types';
+
+/** TODO: docs */
+export interface UiAmountToAmountInstructionData {
+    instruction: TokenInstruction.UiAmountToAmount;
+    amount: Uint8Array;
+}
+
+/** TODO: docs */
+
+/**
+ * Construct a UiAmountToAmount instruction
+ *
+ * @param mint         Public key of the mint
+ * @param amount       UiAmount of tokens to be converted to Amount
+ * @param programId    SPL Token program account
+ *
+ * @return Instruction to add to a transaction
+ */
+export function createUiAmountToAmountInstruction(
+    mint: PublicKey,
+    amount: string,
+    programId = TOKEN_PROGRAM_ID
+): TransactionInstruction {
+    const keys = [{ pubkey: mint, isSigner: false, isWritable: false }];
+    const buf = Buffer.from(amount, 'utf8');
+    const uiAmountToAmountInstructionData = struct<UiAmountToAmountInstructionData>([
+        u8('instruction'),
+        blob(buf.length, 'amount'),
+    ]);
+    const data = Buffer.alloc(uiAmountToAmountInstructionData.span);
+    uiAmountToAmountInstructionData.encode(
+        {
+            instruction: TokenInstruction.UiAmountToAmount,
+            amount: buf,
+        },
+        data
+    );
+    return new TransactionInstruction({ keys, programId, data });
+}
+
+/** A decoded, valid UiAmountToAmount instruction */
+export interface DecodedUiAmountToAmountInstruction {
+    programId: PublicKey;
+    keys: {
+        mint: AccountMeta;
+    };
+    data: {
+        instruction: TokenInstruction.UiAmountToAmount;
+        amount: Uint8Array;
+    };
+}
+
+/**
+ * Decode a UiAmountToAmount instruction and validate it
+ *
+ * @param instruction Transaction instruction to decode
+ * @param programId   SPL Token program account
+ *
+ * @return Decoded, valid instruction
+ */
+export function decodeUiAmountToAmountInstruction(
+    instruction: TransactionInstruction,
+    programId = TOKEN_PROGRAM_ID
+): DecodedUiAmountToAmountInstruction {
+    if (!instruction.programId.equals(programId)) throw new TokenInvalidInstructionProgramError();
+    const uiAmountToAmountInstructionData = struct<UiAmountToAmountInstructionData>([
+        u8('instruction'),
+        blob(instruction.data.length - 1, 'amount'),
```

Comment:
Since length would be depending on the size of amount string , for decoding I assigned it data.length -1 , since u8 instruction would be taking 1 size, i checked it's if it's decoding correctly or not, is it fine or is there some better way for this